src/journal/JournalRecorder.cc

```diff
@@ -201,6 +213,7 @@ void JournalRecorder::open_object_set() {
       create_next_object_recorder(object_recorder);
```

Comment:
Add an "else" branch that will unlock the object_recorder if a create wasn't required --- then ```create_next_object_recorder``` can become ```create_next_object_recorder_unlock``` and ```ObjectRecord::append(..., bool unblock)``` can be removed (merged) and all the extra ```if (unlock)``` conditions can be removed.